pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -235,6 +235,12 @@ message CommandConnect {
     optional string original_auth_data = 8;
     optional string original_auth_method = 9;
 
+    // Feature flags
+    optional FeatureFlags feature_flags = 10;
+}
+
+message FeatureFlags {
```

Comment:
why a feature flag rather than bumping the version? Because some auth will not support reauth?

Reply:
Bumping the version is a very primitive way of checking the supported features. It worked well when the client implementation was only Java and started to be problematic after that. 

The reality is that all the clients out there will have a different set of supported features. Also, if someone just updates the`PulsarApi.proto` it looks like that client is now at the latest version, while in reality it might not be able to understand a specific command.

Exchanging the list of supported features will allow client to selectively implement features.

Reply:
adding `FeatureFlags` makes sense. 👍